hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/HttpServletUtils.java

```diff
@@ -0,0 +1,211 @@
+    case JSON:
+      Map<String, String> errorMap = new HashMap<>();
+      errorMap.put("error", errorMessage);
+      writer.write(JsonUtils.toJsonString(errorMap));
+      break;
+    case XML:
+      writeXmlError(errorMessage, writer);
+      break;
+    default:
+      throw new IOException("Unsupported response format for error response: " + format,
+          new IllegalArgumentException("Bad format: " + format));
+    }
+  }
+
+  private static void writeXmlError(String errorMessage, Writer out) throws IOException {
+    try {
+      DocumentBuilder builder = DOCUMENT_BUILDER_FACTORY.get().newDocumentBuilder();
+      Document doc = builder.newDocument();
+
+      Element root = doc.createElement("error");
+      root.setTextContent(errorMessage);
+      doc.appendChild(root);
+
+      TransformerFactory transformerFactory = TransformerFactory.newInstance();
+      Transformer transformer = transformerFactory.newTransformer();
+      transformer.setOutputProperty(OutputKeys.ENCODING, "UTF-8");
+      transformer.setOutputProperty(OutputKeys.STANDALONE, "no");
+
+      DOMSource source = new DOMSource(doc);
+      StreamResult result = new StreamResult(out);
+      transformer.transform(source, result);
+    } catch (ParserConfigurationException | TransformerException e) {
+      throw new IOException("Failed to write XML error response", e);
+    }
+  }
+
+  /**
+   * Write response according to the specified format.
+   * The caller provides a callback to write the content.
+   *
+   * @param response        the HTTP servlet response
+   * @param format          the response format
+   * @param contentWriter   callback to write content to the writer
+   * @param exceptionClass  class of exception to propagate from contentWriter
+   * @param <E>             the type of exception that may be thrown by contentWriter
+   * @throws IOException if an I/O error occurs
+   * @throws E           if contentWriter throws an exception of type E
+   */
+  public static <E extends Exception> void writeResponse(HttpServletResponse response, ResponseFormat format,
+      CheckedConsumer<Writer> contentWriter, Class<E> exceptionClass) throws IOException, E {
+    response.setContentType(format.getContentType());
+    Writer out = response.getWriter();
+    try {
+      contentWriter.accept(out);
+    } catch (IOException e) {
+      // Always rethrow IOException as-is
+      throw e;
+    } catch (Exception e) {
+      // If exception matches the generic type, throw it
+      if (exceptionClass.isInstance(e)) {
+        throw exceptionClass.cast(e);
+      }
+      // Otherwise wrap in IOException
+      throw new IOException("Failed to write response", e);
+    }
+  }
+
+  /**
+   * Functional interface for operations that accept a parameter and may throw exceptions.
+   *
+   * @param <T> the type of the input to the operation
+   */
+  @FunctionalInterface
+  public interface CheckedConsumer<T> {
+    void accept(T t) throws Exception;
+  }
```

Comment:
In part 2, I think we should reuse `CheckedConsumer` from `org.apache.ratis.util.function` instead of defining new interface.